From 691c5065623a526aa8af664e10f5aae6778eb019 Mon Sep 17 00:00:00 2001
From: Thomas Sibley <trs@bestpractical.com>
Date: Mon, 1 Apr 2013 15:39:23 -0700
Subject: [PATCH] Allow the NOT IN operator in Limit calls

IN/NOT IN is supported by DBIx::SearchBuilder with the 1.63_01 release.
---
 lib/RT/SearchBuilder.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/RT/SearchBuilder.pm b/lib/RT/SearchBuilder.pm
index 1df5ac1..0ace421 100644
--- a/lib/RT/SearchBuilder.pm
+++ b/lib/RT/SearchBuilder.pm
@@ -279,7 +279,7 @@ sub Limit {
                                   |(NOT\s*)?(STARTS|ENDS)WITH
                                   |(NOT\s*)?MATCHES
                                   |IS(\s*NOT)?
-                                  |IN
+                                  |(NOT\s*)?IN
                                   |\@\@)$/ix) {
         $RT::Logger->crit("Possible SQL injection attack: $ARGS{FIELD} $ARGS{OPERATOR}");
         $self->SUPER::Limit(
-- 
1.7.11.3

